### DISCUSSION TITLE
ARM Processor but ARCH linux

### DISCUSSION THREAD

**Discussion by Troodon7** [aircrack-ng/aircrack-ng#2548] [Q&A] [answered]
I did a little bit of digging and have been coming up short on finding support for aircrack-ng for this specific combination in my title, ARM processor but ARCH linux. From what I noticed so far there is ARM support but I hit a bit of a wall when I realized it was only Debian, from what I have found so far:

- linux/arm/v5 (base image: debian:unstable-slim)
- linux/arm/v6 (base image: alpine:3)
- linux/arm/v7 (base image: debian:unstable-slim)

So my question really boils down to is this even currently possible? 

The reasoning for wanting this specific combination to work is because I am working on a project for myself mostly but the community also because the project is a Steam Deck with the original OS so I can keep the gaming functionality but also have all the great tools that I need sometimes for pen-testing purposes for my work and side business. The portability of the deck provides a great on the go solution, the goal would be to eventually build out a repo of easily downloadable tools from here for the deck specifically but I suppose also to a Pi user that wants Arch.

**Comment by Mister-X-** [comment, marked as answer]
I used debian for consistency as it was the one that offered the most architectures. Arch Linux should work just fine as well, and the script called in the Dockerfile handles it (if you look in the workflows, one of them tests arch linux).

If there are issues compiling it on arch, open a bug report.

**Reply by Troodon7** [reply]
I will try this later on and mark as answer if that's the case. Thank you for the response.

**Comment by Troodon7** [comment]
Writing this reply for future readers:

SteamOS is dumb, apparently does not like you installing packages not normally installed, HOWEVER, this can be resolved by running: "sudo steamos-readonly disable" 

then simply install aircrack-ng as you normally would on Arch: "sudo pacman -S aircrack-ng" 

Note there may or may not be a requirement to compile it manually but I did not have any issues. To do this you need to install "autoconf" which is not preinstalled on SteamOS. Then simply follow this nice guide aircrack-ng made: https://www.aircrack-ng.org/doku.php?id=install_aircrack

Last note, once you upgrade your SteamOS there is a slight chance that it will remove everything you installed manually according to some on the SteamDeck Forums. I cannot confirm if this is true at this time but I will say that I've installed other things and so far so good previously. Either way it might be a good idea to make yourself a "setup" script so to speak and keep notes of what you've installed just in case.
